### PR TITLE
refactor: Replace react-bootstrap MenuItems with Antd Menu

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR is first of the series - the goal is to get rid of all react-bootstrap's MenuItem usages. In this PR I refactored CopyToClipboard, LanguagePicker, HeaderActionsDropdown and DateFilterControl. In the case of DateFilterControl, I also fixed styling bug (as on the screenshot below) and replaced Input component for UI consistency.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
DateFilterControl - before:
image
DateFilterControl - after:
image

HeaderActionsDropdown - before:
image
HeaderActionsDropdown - after:
image

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/incubator-superset/issues/11492
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by kgabryje** [comment]
@rusackas @etr2460 Can you take a look?
I just noticed that I forgot about uppercasing in DateFilterControl... but do we actually want it? In my opinion it looks a bit nicer now (see the first and second screenshot).

**Comment by rusackas** [comment]
Just a note that https://github.com/apache/incubator-superset/pull/11491 may affect this, or vice versa.

**Comment by kgabryje** [comment]
@rusackas I rebased it to use changes from https://github.com/apache/incubator-superset/pull/11491

File: superset-frontend/src/common/components/index.tsx [hunk 1]
```diff
@@ -43,6 +45,11 @@ export const MenuItem = styled(AntdMenu.Item)`
   > a {
     text-decoration: none;
   }
+
+  &.ant-menu-item {
+    height: 30px;
+    line-height: 30px;
```
Comment (rusackas) [inline superset-frontend/src/common/components/index.tsx:51]: I guess it was another PR with the same CSS where we were discussing using the Theme's typography size for this... same comment applies here :)

**Review by rusackas — APPROVED** [review]
One remaining nit here about the 30px value we discussed on another PR, but in general this LGTM. Approving, but not yet merging in case you'd like to tackle that tweak here, or rebase if the other PR comes in first.

File: superset-frontend/src/common/components/index.tsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/src/common/components/index.tsx:51]: Done

**Pull request merged**